src/App.tsx

```diff
@@ -1,14 +1,23 @@
 import React from 'react';
+import posts from './api/posts';
+import comments from './api/comments';
+import users from './api/users';
+import { PostList } from './components/PostList';
 
 import './App.scss';
 
-// import posts from './api/posts';
-// import comments from './api/comments';
-// import users from './api/users';
+const preparedPosts: Post[] = posts.map(post => ({
+  ...post,
+  user: users.find(user => user.id === post.userId),
+  comments: comments.filter(comment => comment.postId === post.id) as Comment[],
```

Comment:
You don't really need to set type here explicitly cause you have already defined `comments` type
```suggestion
  comments: comments.filter(comment => comment.postId === post.id),
```

Reply:
I know, but for some reason, it doesn't work. Then Dima told me to do it in this way at the Q&A